src/algo/ArrayList.js

```diff
@@ -260,6 +263,8 @@ export default class ArrayList extends Algorithm {
 		this.removeIndexCodeID = [];
 
 		this.resetIndex = this.nextIndex;
+		this.infoLabelID = this.nextIndex++;
+		this.cmd(act.createLabel, this.infoLabelID, '', INFO_MSG_X, INFO_MSG_Y, 0);
```

Comment:
```suggestion
		this.infoLabelID = this.nextIndex++;
		this.cmd(act.createLabel, this.infoLabelID, '', INFO_MSG_X, INFO_MSG_Y, 0);
		this.resetIndex = this.nextIndex;
```
Same as CSLL, but we want to define the infoLabel before setting this.resetIndex. This way the infoLabelID persists even when the animation is reset.